### DISCUSSION TITLE
is `loadDefaultProcessPlanners` needed?

### DISCUSSION THREAD

**Discussion by mohamedsayed18** [tesseract-robotics/tesseract_ros#120] [Q&A] [answered]
In the puzzle_piece_auxillary_axes_example, Is it need to use the `loadDefaultProcessPlannners` method as [here](https://github.com/ros-industrial-consortium/tesseract_ros/blob/ee901eeea47cd5773d8055ad838ac81598470cea/tesseract_ros_examples/src/puzzle_piece_auxillary_axes_example.cpp#L216) ?. 
Because [here](https://github.com/ros-industrial-consortium/tesseract_ros/blob/ee901eeea47cd5773d8055ad838ac81598470cea/tesseract_ros_examples/src/puzzle_piece_auxillary_axes_example.cpp#L224) we added a new planner
Do the planners override each other or they are working together?
In case if they are working together can it happen that a planner has parameters that are contrary to parameters in other planners?

**Comment by Levi-Armstrong** [comment, marked as answer]
> In the puzzle_piece_auxillary_axes_example, Is it need to use the loadDefaultProcessPlannners method as here ?.

In this case, since the only process planner being used is the one added in the example it is not necessary to call `loadDefaultProcessPlannners`.

> Do the planners override each other or they are working together?

It stores them in a map so if you add a planner with the same name as one that is already stored it will be replaced.

> In case if they are working together can it happen that a planner has parameters that are contrary to parameters in other planners?

Currently, the process planners do not interact, but the low-level planners like Descartes, TrajOpt, and OMPL do. It is possible to set up parameter for the individual planners which are not compatible. For example, if you use the raster planner where the raster (Cartesian Planner) has a contact distance threshold of 1cm but for the transitions (Freespace Planner) you set a contact threshold of 2cm this could cause it to fail because the start and end of the transition may violation the 2cm distance because they were planned with the 1cm distance threshold.

**Comment by mohamedsayed18** [comment]
Thank you, I have another question to make sure I'm understanding how it works
For each PlanInstruction can I choose a specific Planner?
Or if two planners were chosen their parameters will be applied to all the PlanInstrucions in the program
